app/validators/claim/base_claim_validator.rb

```diff
@@ -260,28 +258,26 @@ def validate_retrial_length(field)
   end
 
   def validate_trial_actual_length_consistency
-    return unless actual_length_consistency_for_trial
-
-    # As we are using Date objects without time information, we loose precision, so adding 1 day will workaround this.
-    return unless ((@record.trial_concluded_at - @record.first_day_of_trial).days + 1.day) <
-                  @record.actual_trial_length.days
+    return unless actual_length_consistency_for(requires_trial_dates?, @record.actual_trial_length, @record.first_day_of_trial, @record.trial_concluded_at)
     add_error(:actual_trial_length, 'too_long')
   end
 
-  def actual_length_consistency_for_trial
-    requires_trial_dates? &&
-      @record.actual_trial_length.present? &&
-      @record.first_day_of_trial.present? &&
-      @record.trial_concluded_at.present?
+  def validate_retrial_actual_length_consistency
+    return unless actual_length_consistency_for(requires_retrial_dates?, @record.retrial_actual_length, @record.retrial_started_at, @record.retrial_concluded_at)
+    add_error(:retrial_actual_length, 'too_long')
   end
 
-  def validate_retrial_actual_length_consistency
-    return unless actual_length_consistency_for_retrial
+  def actual_length_consistency_for(requires_dates, actual_length, start_date, end_date)
```

Comment:
naming suggestion: `actual_length_consistent?` | `actual_length_valid?`

which matches the other method `trial_length_valid?` lower down called by this method

Reply:
Good call on the renamings  "naming things is hard"! :;